ir/state.cpp

```diff
@@ -95,14 +123,27 @@ const StateValue& State::getAndAddUndefs(const Value &val) {
   return v;
 }
 
+const StateValue& State::getAndAddPoisonUB(const Value &val) {
+  auto &v = (*this)[val];
+  if (v.isValid()) {
+    // If val is an aggregate, all elements should be non-poison
+    expr is_np = v.non_poison.isBool() ? v.non_poison : v.non_poison == 0;
+    if (!is_np.isTrue()) {
+      analysis.non_poison_vals.insert(&val);
+      addUB(move(is_np));
+    }
+  }
+  return v;
```

Comment:
this is not resetting np to true if np wasn't true before.

Reply:
Callers of getAndAddPoisonUB are using the `value` field only (except BinOp, which assumes non_poison is true if if was div or rem so now).
Or, the state value can be temporarily stored to `tmp_values[i_tmp_values++]`, its non_poison is updated to true, and the reference may be returned, but it would cause use of an extra slot.

Reply:
it's alright to use an extra slot

Reply:
Well, okay then. I will update this part